From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nick Chan <towinchenmi@gmail.com>
Date: Sun, 1 Jun 2025 22:21:11 +0800
Subject: [PATCH 06/10] of: Export of_dma_range_parser_init() alias

Export of_dma_range_parser_init() alias to allow non-PCI drivers to sanely
parse dma-ranges.

Signed-off-by: Nick Chan <towinchenmi@gmail.com>
---
 drivers/of/address.c       | 1 -
 include/linux/of_address.h | 1 +
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/of/address.c b/drivers/of/address.c
index 499d37cea..60c68e87f 100644
--- a/drivers/of/address.c
+++ b/drivers/of/address.c
@@ -793,7 +793,6 @@ int of_pci_dma_range_parser_init(struct of_pci_range_parser *parser,
 	return parser_init(parser, node, "dma-ranges");
 }
 EXPORT_SYMBOL_GPL(of_pci_dma_range_parser_init);
-#define of_dma_range_parser_init of_pci_dma_range_parser_init
 
 struct of_pci_range *of_pci_range_parser_one(struct of_pci_range_parser *parser,
 						struct of_pci_range *range)
diff --git a/include/linux/of_address.h b/include/linux/of_address.h
index 0cff90365..37f59a521 100644
--- a/include/linux/of_address.h
+++ b/include/linux/of_address.h
@@ -174,6 +174,7 @@ static inline void __iomem *of_iomap(struct device_node *device, int index)
 	return NULL;
 }
 #endif
+#define of_dma_range_parser_init of_pci_dma_range_parser_init
 #define of_range_parser_init of_pci_range_parser_init
 
 static inline const __be32 *of_get_address(struct device_node *dev, int index,
